socorro/external/boto/connection_context.py

```diff
@@ -2,38 +2,72 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import contextlib
-import datetime
-import json
-import socket
+import io
+import logging
+import random
 import time
 
-import boto
-import boto.exception
-import boto.s3.connection
-from configman import Namespace, RequiredConfig, class_converter
-from configman.converters import str_to_boolean
-import markus
+import boto3
+from botocore.client import ClientError, Config
+from configman import Namespace, RequiredConfig
 
-from socorro.lib.ooid import date_from_ooid
+from socorro.lib.util import retry
+
+
+logger = logging.getLogger(__name__)
+
+
+def wait_times_connect():
+    """Return generator for wait times between failed connection attempts.
+
+    We have this problem where we're binding IAM credentials to the EC2 node
+    and on startup when boto3 goes to get the credentials, it fails for some
+    reason and then degrades to hitting the https://s3..amazonaws.net/
+    endpoint and then fails because that's not a valid endpoint.
+
+    This sequence increases the wait times and adds some jitter.
+
+    """
+    for i in [5, 5, 5, 5, 5]:
+        yield i + random.uniform(-2, 2)  # nosec
```

Comment:
what does `# nosec` do?

Reply:
I sometimes run `bandit` on the code. The `nosec` tells bandit that, yes, this is a call to random, but it's ok and not a security issue.